CMSgov/cms-carts-seds
# MDCT-202: Basic State Info

# Description

[MDCT-202: Basic State Info](https://qmacbis.atlassian.net/browse/MDCT-202)

This story introduces the functionality to view and update the Basic State Information section (among other sections but the scope of this story is limited to Basic State Information).

List of changes:
* Creates `SectionTable` and associated seed data.
* Creates `SectionBaseTable` and associated seed data (this isn't used anywhere yet).
* Creates the `/section/{year}/{state}` endpoint to fetch the associated sections of a given report.
* Creates the `update_report/{year}/{state}` endpoint that updates the associated sections of a given report.
* Updates the UI to use these new endpoints.
* Removes some of the measures boilerplate.

## How to test

Manual testing:
* Check out the branch
* Update your `.env` file with the two fields added to `.env-example`
* run `./dev local`
* Navigate to a report by clicking the `Edit` option on the dashboard.
* Ensure the report loads
* Try editing a field and ensure it saves
* Ensure the state status updates from `Not started` to `In progress` back on the dashboard

Run the unit tests
* `cd` to `services/app-api`
* run `yarn test`

## Dependencies 

N/A

# Get it done

Now that the PR is open this is what happens next

## Code authors checklist

## Reviewers Checklist (Two different people)

## Assignee
